src/nixos.rs

```diff
@@ -22,15 +22,16 @@ impl interface::OsArgs {
     pub fn run(self) -> Result<()> {
         use OsRebuildVariant::*;
         match self.subcommand {
-            OsSubcommand::Boot(args) => args.rebuild(Boot),
-            OsSubcommand::Test(args) => args.rebuild(Test),
-            OsSubcommand::Switch(args) => args.rebuild(Switch),
+            OsSubcommand::Boot(args) => args.rebuild(Boot, None),
+            OsSubcommand::Test(args) => args.rebuild(Test, None),
+            OsSubcommand::Switch(args) => args.rebuild(Switch, None),
             OsSubcommand::Build(args) => {
                 if args.common.ask || args.common.dry {
                     warn!("`--ask` and `--dry` have no effect for `nh os build`");
                 }
-                args.rebuild(Build)
+                args.rebuild(Build, None)
             }
+            OsSubcommand::BuildVm(args) => args.build(BuildVm),
```

Comment:
Would it not be put the logic for `BuildVm` in a separate method? This PR opens the door to more robust builds, but I'd like to avoid overloading build each time if possible (which is sort of what's happening in rebuild.)

```rs
impl OsBuildVmArgs {
    pub fn build_vm(self) -> Result<()> {
        let final_attr = get_final_attr(true, self.with_bootloader);
        self.common.rebuild(OsRebuildVariant::BuildVm, Some(final_attr))
    }
}
```

and then

```diff
-    OsSubcommand::BuildVm(args) => args.build(BuildVm),
+    OsSubcommand::BuildVm(args) => args.build_vm(),
```